### PR TITLE
PR#26 (add/getAllKeysStartingWith)

### PR DESCRIPTION
**Pull request by rgadon107**
This PR adds tests for the config store's API function `getAllKeysStartingWith()`.  It also removes the dead code in the function.

## TODO

* [x] Add unit test for when receiving a starts with string and store has configs
* [x]  Add integration tests for same test.
* [x] Add unit test for when no store keys start with the given string
* [x] Add integration tests for same test.
* [x] Add unit test for when the store is empty.
* [x] Add integration tests for same test.

### CODE REVIEW

**Comment by rgadon107** [comment]
@hellofromtonya Well, your commits this morning were _very_ instructive. Not having built a unit test before to mock a loop, the mocks you wrote are useful guidance for future instances of that type of test. 

Thanks for creating the TODO list above. That's a useful tool going forward, not just on this branch, but future branches as well.

File: mu-plugins/central-hub/src/config-store/api.php [hunk 1]
```diff
@@ -126,5 +126,11 @@ function getAllKeysStartingWith( $starts_with ) {
 		}
 	}
 
+	if ( empty( $filtered_keys ) ) {
+		throw new \InvalidArgumentException(
```
Comment (hellofromtonya) [inline mu-plugins/central-hub/src/config-store/api.php]: @rgadon107 I don't think this is the right response.  I think the severity of what to do when a configuration doesn't exit belongs to the code that calls this function.  For example, maybe there aren't any metaboxes or custom post types to be registered.  In that case, it's up to the Metabox or CPT code to decide if shutting down the website with a fatal error is the right response.  Don't you agree?

Comment (rgadon107) [reply mu-plugins/central-hub/src/config-store/api.php]: @hellofromtonya I see your point, and it makes sense. Let me go back, retrace the code, and think about the context in which the API function `getAllKeysStartingWith` is used.

File: mu-plugins/central-hub/tests/phpunit/integration/config-store/getAllTestsStartingWith.php [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+			],
+		];
+		foreach ( $configs as $store_key => $config_to_store ) {
+			loadConfig( $store_key, $config_to_store );
+		}
+
+		$this->assertSame( [ 'metabox.events', 'metabox.members' ], getAllKeysStartingWith( 'metabox.' ) );
+		$this->assertSame( [ 'shortcode.qa' ], getAllKeysStartingWith( 'shortcode.' ) );
+		$this->assertSame( [ 'custom_post_type.books' ], getAllKeysStartingWith( 'custom_post_type.' ) );
+	}
```
Comment (hellofromtonya) [inline mu-plugins/central-hub/tests/phpunit/integration/config-store/getAllTestsStartingWith.php]: @rgadon107 You want to clean up at the end of this particular test to remove the configs that were added by this test.  Best Practice: clean as you go.

File: mu-plugins/central-hub/tests/phpunit/integration/config-store/getAllTestsStartingWith.php [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+	public function test_should_return_empty_array_when_filtered_keys_do_not_start_with_a_given_string() {
+		$expected = [];
+		$this->assertSame( $expected, getAllKeysStartingWith( 'taxonomy.' ) );
+	}
+
+	/**
+	 * Test getAllKeysStartingWith() should return an empty array when the store is empty.
+	 */
+	public function test_should_return_empty_array_when_the_store_is_empty() {
+		self::setUpBeforeClass();
```
Comment (hellofromtonya) [inline mu-plugins/central-hub/tests/phpunit/integration/config-store/getAllTestsStartingWith.php]: @rgadon107 This line of code isn’t needed once the above test cleans up after itself.

File: mu-plugins/central-hub/tests/phpunit/integration/config-store/getAllTestsStartingWith.php (hunk 2 above)
Comment (rgadon107) [reply mu-plugins/central-hub/tests/phpunit/integration/config-store/getAllTestsStartingWith.php]: @hellofromtonya Point well taken. Please see commit [0dc219f](https://github.com/rgadon107/cornerstone/pull/26/commits/0dc219f1b37f967450e291049a9081b0a363e0a0).

File: mu-plugins/central-hub/tests/phpunit/integration/config-store/getAllTestsStartingWith.php (hunk 3 above)
Comment (rgadon107) [reply mu-plugins/central-hub/tests/phpunit/integration/config-store/getAllTestsStartingWith.php]: @hellofromtonya Done. Per commit [0dc219f](https://github.com/rgadon107/cornerstone/pull/26/commits/0dc219f1b37f967450e291049a9081b0a363e0a0).

File: mu-plugins/central-hub/tests/phpunit/integration/config-store/getAllTestsStartingWith.php (hunk 2 above)
Comment (hellofromtonya) [reply mu-plugins/central-hub/tests/phpunit/integration/config-store/getAllTestsStartingWith.php]: @rgadon107 Actually, you know the keys that were loaded from `$configs`. Loop through those and remove.  It’s better not to invoke `setUpBeforeClass()`.  Why?  It’s purpose is to set up the class itself, ie before the tests run.

**Comment by rgadon107** [comment]
@hellofromtonya Per your last comment, please see commit  [53c024c](https://github.com/rgadon107/cornerstone/pull/26/commits/53c024c157aaecb05f529ba4047990828bd5eece).

File: mu-plugins/central-hub/tests/phpunit/integration/config-store/getAllTestsStartingWith.php (hunk 2 above)
Comment (hellofromtonya) [reply mu-plugins/central-hub/tests/phpunit/integration/config-store/getAllTestsStartingWith.php]: @rgadon107 I moved this code, combined it, and refactored into a separate static function that can be reused.
